consumer/statistics/tracker.go

```diff
@@ -69,17 +70,36 @@ func (sst *SessionStatisticsTracker) markSessionEnd() {
 	sst.sessionStart = nil
 }
 
+// calcStatDiff takes in the old and the new values of statistics, returns the calculated delta
+func (sst *SessionStatisticsTracker) calcStatDiff(old, new uint64) (res uint64) {
+	if old > new {
+		return new
+	}
+	return new - old
+}
+
+// GetStatisticsDiff calculates the difference in bytes between the old stats and new
+func (sst *SessionStatisticsTracker) GetStatisticsDiff(old, new consumer.SessionStatistics) consumer.SessionStatistics {
+	return consumer.SessionStatistics{
+		BytesSent:     sst.calcStatDiff(old.BytesSent, new.BytesSent),
+		BytesReceived: sst.calcStatDiff(old.BytesReceived, new.BytesReceived),
+	}
+}
+
 // ConsumeStatisticsEvent handles the connection statistics changes
 func (sst *SessionStatisticsTracker) ConsumeStatisticsEvent(stats consumer.SessionStatistics) {
-	sst.sessionStats = stats
+	diff := sst.GetStatisticsDiff(sst.lastStats, stats)
+	sst.sessionStats.BytesReceived += diff.BytesReceived
```

Comment:
If you are making a function to calc diff, maybe it's worth making a function newStats := AddUp(sessionStats, diff)

Reply:
Those diff and add methods can even be part of SessionStatistics since we are doing calculations with them anyway